server/server/app.py

```diff
@@ -11,22 +12,80 @@
 back_button = "</br></br><a href='/'>Back</a>"
 
 
+def query_decorator(func):
+    def wrapper(*args, **kwargs):
+
+        try:
+
+            ret_dict = func(*args, **kwargs)
+
+            if not ret_dict["data"]:
+                ret_dict['status_code'] = 500
```

Comment:
this is a bug. if `"data"` not it `ret_dict` you will get a `KeyError` Exception

You could do either:
`if "data" not in ret_dict:`
or:
```
data = ret_dict.get("data")
if not data:
```

I would go with Option 2 since you need `data` a few lines later anyways